### PR TITLE
Introduced Cavium target

### PR DESCRIPTION
**Pull request by rspolyak**
Added Cavium target. This target is used to build linux kernel with vendor specific patches (set by LINUX_PATCH). This change does not impact existing vendors.

### CODE REVIEW

File: src/Makefile [hunk 1]
```diff
@@ -61,7 +61,7 @@ mlnx/swss_1.0.0_amd64.deb mlnx/intfsyncd mlnx/neighsyncd mlnx/orchagent mlnx/por
 	cp sonic-swss/debian/swss/usr/bin/fpmsyncd .
 	
 $(addprefix sonic-linux-kernel/,linux-headers-3.16.0-4-amd64_3.16.7-ckt11-2+acs8u2_amd64.deb linux-headers-3.16.0-4-common_3.16.7-ckt11-2+acs8u2_amd64.deb linux-image-3.16.0-4-amd64-dbg_3.16.7-ckt11-2+acs8u2_amd64.deb linux-image-3.16.0-4-amd64_3.16.7-ckt11-2+acs8u2_amd64.deb xen-linux-system-3.16.0-4-amd64_3.16.7-ckt11-2+acs8u2_amd64.deb):
-	pushd sonic-linux-kernel; sudo ./build.sh; popd
+	pushd sonic-linux-kernel; sudo ./build.sh $(LINUX_PATCH); popd
```
Comment (qiluo-msft) [inline src/Makefile]: Have you modified build.sh?

**Comment by qiluo-msft** [comment]
Relating to https://github.com/Azure/sonic-linux-kernel/pull/3

**Comment by lguohan** [comment]
this has conflict with current master? can you update to current master?

**Comment by garyachy** [comment]
@stcheng and @lguohan please review

**Comment by lguohan** [comment]
@garyachy , can you bring this commit the the latest master? This branch has conflicts and must be resolved.

**Comment by rspolyak** [comment]
@ lguohan sure, will do it soon

**Comment by rspolyak** [comment]
@lguohan please review

File: Makefile [hunk 2]
```diff
@@ -113,3 +118,5 @@ brcm-all: target/sonic-generic.bin $(addprefix target/,docker-syncd.gz docker-or
 
 ## Note: docker-fpm.gz must be the last to build the implicit dependency fpmsyncd
 mlnx-all: target/sonic-generic.bin $(addprefix target/,docker-syncd-mlnx.gz docker-orchagent-mlnx.gz docker-fpm.gz docker-database.gz)
+
+cavm-all: target/sonic-generic.bin
```
Comment (stcheng) [inline Makefile]: Doesn't need to have other dockers like the above two?

**Comment by rspolyak** [comment]
Dockers are introduced via another PR

Sent from my phone

15 ???. 2016 ?. ? 11:04 Shuotian Cheng <notifications@github.com<mailto:notifications@github.com>> ???????(??):


@stcheng commented on this pull request.

________________________________

In Makefile<https://github.com/Azure/sonic-buildimage/pull/4#pullrequestreview-111589>:

> @@ -113,3 +118,5 @@ brcm-all: target/sonic-generic.bin $(addprefix target/,docker-syncd.gz docker-or

 ## Note: docker-fpm.gz must be the last to build the implicit dependency fpmsyncd
 mlnx-all: target/sonic-generic.bin $(addprefix target/,docker-syncd-mlnx.gz docker-orchagent-mlnx.gz docker-fpm.gz docker-database.gz)
+
+cavm-all: target/sonic-generic.bin


Doesn't need to have other dockers like the above two?

-

**Comment by rspolyak** [comment]
Dockers for Cavium :

https://github.com/Azure/sonic-buildimage/pull/5/files

File: installer/x86_64/platforms/x86_64-accton_as7512_32x-r0 [hunk 3]
```diff
@@ -0,0 +1,3 @@
+CONSOLE_PORT=0x2f8
+CONSOLE_DEV=1
+CONSOLE_SPEED=115200
```
Comment (lguohan) [inline installer/x86_64/platforms/x86_64-accton_as7512_32x-r0]: all sonic kernel use 9600 baud rate

**Comment by lguohan** [comment]
as noted in the comment in Azure/sonic-linux-kernel#3, I suggest drop the LINUX_PATCH for the sonic image build for now. We should be able to merge the other parts

File: src/Makefile (hunk 1 above)
Comment (lguohan) [inline src/Makefile]: suggest drop this for now. Let's first agree on how the sonic kernel is going to be build in the other repo.

File: Makefile [hunk 4]
```diff
@@ -7,6 +7,11 @@ PASSWORD_ENCRYPTED=
 ## Select bash for commands
 SHELL := /bin/bash
 
+## Path to vendor linux patch folder
+ifeq ($(MAKECMDGOALS),cavm-all)
+export LINUX_PATCH=cavm-as7512
+endif
```
Comment (lguohan) [inline Makefile]: suggest drop this for now. Let's first agree on how the sonic kernel is going to be build in the other repo

**Pull request closed without merging**